### ISSUE TITLE
[BUG] `verify-container-capabilities` flags if forbidden capabilities are not dropped

### ISSUE THREAD

**Issue by cailyoung** [stackrox/kube-linter#288]
**System info:**
 - OS: Ubuntu in a GHA runner

**Describe the bug**
I have a custom check which uses this config:
```yaml
checks:
  doNotAutoAddDefaults: true
  
customChecks:
  - name: no-sys-admin-capability
    template: verify-container-capabilities
    params:
      forbiddenCapabilities:
        - SYS_ADMIN
    remediation: Do not use SYS_ADMIN as a capability - this is incredibly insecure, please set a less-permissive security context.
```

I'm getting this lint failure:
`container "delete-db-properties" has DROP capabilities: ["NET_RAW"], but does not drop capability "SYS_ADMIN" which is required `

Perhaps it's a documentation gap; but I would expect this sort of check config to only cause a lint failure if the capability is in the `add:` section, not if it _isn't in the `drop:` section_.

**To Reproduce**

Run kubelinter with the above config against a helm template with this kind of `securityContext`:

```yaml
        securityContext:
          runAsNonRoot: true
          allowPrivilegeEscalation: false
          capabilities:
            drop:
              - NET_RAW
```

**Expected behavior**
This check should only fail for `forbiddenCapabilities` if they are explicitly added, not if they _aren't dropped_.

**Comment by viswajithiii** [comment]
Hi @cailyoung, thanks for filing this issue!

The reason we went with this approach is that the default capabilities are set at the container runtime level, and we have no guarantee on what that will be set to at the point where `kube-linter` runs. So, if a capability is not specified in `DROP`, we cannot be sure that it is off, since it might be part of the default set of capabilities.

I guess one solution we could adopt is to add another optional parameter to this check template that lets users specify the default capabilities.

**Comment by cailyoung** [comment]
👋 Hey there, thanks for the quick reply!

Your reasoning makes sense - perhaps if you could enable a custom check writer to supply something like:

```yaml
customChecks:
  - name: no-sys-admin-capability
    template: verify-container-capabilities
    params:
      mustNotAddCapabilities:
        - SYS_ADMIN
      mustDropCapabilities:
        - FOO
      mustNotDropCapabilities:
        - BAZ
    remediation: Do not use SYS_ADMIN as a capability - this is incredibly insecure, please set a less-permissive security context.
```

It would allow enough flexibility here for my case. I think that as the check author I'm familiar enough with my context to understand whether it's safe to make assumptions about what the runtime is defaulting, does that seem reasonable?
